src/Framework/Blocks/BlockCollection.php

```diff
@@ -79,24 +79,117 @@ public function getBlocks(): array
     /**
      * @unreleased
      *
-     * @param BlockModel[] $blocks
-     * @param string $name
-     * @return BlockModel|null
+     * @return BlockCollection|null
      */
-    public function findByName(string $name, array $blocks = null)
+    public function findByName(string $blockName, int $blockIndex = 0, string $return = 'self')
```

Comment:
When do you anticipate needing to use this return param?

Reply:
since we're using the `findByNameRecursive` method for those now, do we still need that here for the public method?  I'm just trying to think of a situation where you would want to use `findByName` and return the parent collection vs just simply returning the block.

Reply:
Gotcha!

Reply:
Resolved 87d70a6e2644711c2028fca0e50bdc0c9aa86576